infra/services/sql.bicep

```
@@ -0,0 +1,94 @@
+targetScope = 'resourceGroup'
+
+@minLength(3)
+@maxLength(17)
+@description('The base resource name. SQL Server names have a max length restriction.')
+param baseName string = resourceGroup().name
+
+@description('The location of the resource. By default, this is the same as the resource group.')
+param location string = resourceGroup().location
+
+@description('The client OID to grant access to test resources.')
+param testApplicationOid string
+
+@description('SQL Server administrator login name.')
+param sqlAdminLogin string = 'mcptestadmin'
+
+@description('SQL Server administrator password.')
+@secure()
+param sqlAdminPassword string = newGuid()
+
+// SQL Server resource
+resource sqlServer 'Microsoft.Sql/servers@2023-05-01-preview' = {
+  name: '${baseName}-sql'
```

there's no reason to append anything to base name unless you'll have multiple of the same resource in the deployment

```suggestion
  name: baseName
```